x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_fetch_alerts.tsx

```diff
@@ -0,0 +1,276 @@
+  loading: boolean;
+  request: Omit<FetchAlertsArgs, 'skip'>;
+  response: AlertResponseState;
+}
+
+type AlertActions =
+  | { type: 'loading'; loading: boolean }
+  | { type: 'response'; alerts: EcsFieldsResponse[]; totalAlerts: number }
+  | { type: 'resetPagination' }
+  | { type: 'request'; request: Omit<FetchAlertsArgs, 'skip'> };
+
+const initialAlertState: AlertStateReducer = {
+  loading: true,
+  request: {
+    featureIds: [],
+    fields: [],
+    query: {
+      bool: {},
+    },
+    pagination: {
+      pageIndex: 0,
+      pageSize: 50,
+    },
+    sort: defaultSort,
+  },
+  response: {
+    alerts: [],
+    totalAlerts: -1,
+    isInitializing: true,
+    updatedAt: 0,
+  },
+};
+
+function alertReducer(state: AlertStateReducer, action: AlertActions) {
+  switch (action.type) {
+    case 'loading':
+      return { ...state, loading: action.loading };
+    case 'response':
+      return {
+        ...state,
+        loading: false,
+        response: {
+          isInitializing: false,
+          alerts: action.alerts,
+          totalAlerts: action.totalAlerts,
+          updatedAt: Date.now(),
+        },
+      };
+    case 'resetPagination':
+      return {
+        ...state,
+        request: {
+          ...state.request,
+          pagination: {
+            ...state.request.pagination,
+            pageIndex: 0,
+          },
+        },
+      };
+    case 'request':
+      return { ...state, request: action.request };
+    default:
+      throw new Error();
+  }
+}
+export type UseFetchAlerts = ({
+  featureIds,
+  fields,
+  query,
+  pagination,
+  skip,
+  sort,
+}: FetchAlertsArgs) => [boolean, FetchAlertResp];
+const useFetchAlerts = ({
+  featureIds,
+  fields,
+  query,
+  pagination,
+  skip,
+  sort,
+}: FetchAlertsArgs): [boolean, FetchAlertResp] => {
+  const refetch = useRef<Refetch>(noop);
+  const abortCtrl = useRef(new AbortController());
+  const searchSubscription$ = useRef(new Subscription());
+  const [{ loading, request: alertRequest, response: alertResponse }, dispatch] = useReducer(
+    alertReducer,
+    initialAlertState
+  );
+  const prevAlertRequest = useRef<AlertRequest | null>(null);
+  const inspectQuery = useRef<InspectQuery>({
+    request: [],
+    response: [],
+  });
+  const { data } = useKibana().services;
+
+  const getInspectQuery = useCallback(() => inspectQuery.current, []);
+  const refetchGrid = useCallback(() => {
+    if ((prevAlertRequest.current?.pagination?.pageIndex ?? 0) !== 0) {
+      dispatch({ type: 'resetPagination' });
+    } else {
+      refetch.current();
+    }
+  }, []);
+
+  const fetchAlerts = useCallback(
+    (request: AlertRequest | null) => {
+      if (request == null || skip) {
+        return;
+      }
+
+      const asyncSearch = async () => {
+        prevAlertRequest.current = request;
+        abortCtrl.current = new AbortController();
+        dispatch({ type: 'loading', loading: true });
+        if (data && data.search) {
```

Comment:
is this fetch missing a try/catch for error handling?

Reply:
I do not think so because the error should do the job.